### ISSUE TITLE
S324 false negative on mixed-case hash algorithm names

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#16525]
### Summary

`hashlib.new` interprets the algorithm name case-insensitively. [`hashlib-insecure-hash-function` (S324)](https://docs.astral.sh/ruff/rules/hashlib-insecure-hash-function/) recognizes lowercase and uppercase names, but not mixed-case names.
```console
$ cat >s324.py <<'# EOF'
import hashlib
print(hashlib.new("Md5").hexdigest())
# EOF

$ python s324.py
d41d8cd98f00b204e9800998ecf8427e

$ ruff check --isolated --select S324 s324.py
All checks passed!
```

**Comment by VascoSch92** [comment]
If no one is taking this, I can do it. It should be quick! 😉

**Comment by ntBre** [comment]
I think this was closed by #16552 🎉
